SPEC.md

```diff
@@ -935,3871 +1811,7403 @@ In operations on mismatched numeric types (e.g. `Int` + `Float`), the `Int` type
+```json
+{}
+```
 
-```wdl
-# task foo has an output y
-call foo
-String x = foo.y
+Test config:
 
-Stuct MyType {
-  String b
+```json
+{
+  "fail": true,
+  "return_code": 42
 }
-MyType z = MyType { z: 'hello' }
-String a = z.b
 ```
+</p>
+</details>
 
-#### Ternary operator (if-then-else)
+<details>
+<summary>
+Example: all_return_codes_task.wdl
 
-This is an operator that takes three arguments: a condition expression, an if-true expression, and an if-false expression. The condition is always evaluated. If the condition is true then the if-true value is evaluated and returned. If the condition is false, the if-false expression is evaluated and returned. The if-true and if-false expressions must return values of the same type, such that the value of the if-then-else is the same regardless of which side is evaluated.
+```wdl
+version 1.1
 
-Examples:
+task multi_return_code_task {
+  command <<<
+  exit 42
+  >>>
 
-- Choose whether to say "good morning" or "good afternoon":
-```wdl
-task greet {
-  input {
-    Boolean morning
+  runtime {
+    return_codes: "*"
   }
-  String greeting = "good ~{if morning then "morning" else "afternoon}"
 }
 ```
-- Choose how much memory to use for a task:
-```wdl
-task mytask {
-  input {
-    Array[String] array
-  }
-  Int array_length = length(array)
-  Int memory = if array_length > 100 then "16GB" else "8GB"
-}
+</summary>
+<p>
+Example input:
+
+```json
+{}
 ```
 
-#### Function Calls
+Example output:
 
-WDL provides a [standard library](#standard-library) of functions. These functions can be called using the syntax `func(p1, p2, ...)`, where `func` is the function name and `p1` and `p2` are parameters to the function.
+```json
+{}
+```
 
-#### Expression Placeholders and String Interpolation
+Test config:
 
-Any WDL string expression may contain one or more "placeholders" of the form `~{*expression*}`, each of which contains a single expression. Note that placeholders of the form `${*expression*}` may also be used interchangably, but their use is discouraged for reasons discussed in the [command section](#expression-placeholders) and may be deprecated in a future version of the specification.
+```json
+{
+  "return_code": 42
+}
+```
+</p>
+</details>
 
-When a string expression is evaluated, its placeholders are evaluated first, and their values are then substituted for the placeholders in the containing string.
+#### Reserved `runtime` hints
 
-```wdl
-Int i = 3
-String s = "~{1 + i}"  # s == "4"
-```
+The following attributes are considered "hints" rather than requirements. They are optional for execution engines to support. The purpose of reserving these attributes is to encourage interoperability of tasks and workflows between different execution engines.
 
-As another example, consider how the following expression would be parsed:
+In WDL 2.0, these attributes will move to the new `hints` section.
```

Comment:
we are introducing hints in 1.2, so these attributes will then be made available there will they not?

Reply:
Good catch